lib/src/Pos/Worker/Block.hs

```diff
@@ -211,7 +211,10 @@ dropObftEbb genesisConfig txpConfig = do
     tipHeader <- DB.getTipHeader
     case tipHeader of
         BlockHeaderMain _      -> pure ()
-        BlockHeaderGenesis _   -> do
+        -- If we're starting up a chain in OBFT mode, we need to ensure
+        -- that we don't rollback the actual genesis block out of existence
+        -- (i.e. the EBB at epoch 0):
+        BlockHeaderGenesis _   -> unless (tipHeader ^. epochIndexL == 0) $ do
```

Comment:
Yes, this looks completely reasonable. My memory is fuzzy, but I thought we had tested scenarios where the cluster started in OBFT mode, but perhaps I am misremembering - it seems like we would've hit the same snag that Serge encountered here.

Reply:
@mhuesch: Yeah, we definitely had tested those scenarios before but I think after implementing your rollback solution we eventually ended up accepting a trade-off here such that we'd not be able to start up a chain directly in OBFT.